yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -0,0 +1,354 @@
+package anon
+
+import (
+	"fmt"
+
+	pg_query "github.com/pganalyze/pg_query_go/v6"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/query/queryparser"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	SCHEMA_KIND_PREFIX     = "schema_"
+	TABLE_KIND_PREFIX      = "table_"
+	COLUMN_KIND_PREFIX     = "col_"
+	INDEX_KIND_PREFIX      = "index_"
+	CONSTRAINT_KIND_PREFIX = "constraint_"
+	ALIAS_KIND_PREFIX      = "alias_"
+	TYPE_KIND_PREFIX       = "type_"
+	ROLE_KIND_PREFIX       = "role_"
+	CONST_KIND_PREFIX      = "const_"
+	DEFAULT_KIND_PREFIX    = "anon_"           // fallback for any other identifiers
+	SALT_KEY_METADB        = "anonymizer_salt" // Key to store salt in MetaDB MSR for consistent anonymization across runs
+)
+
+type Anonymizer interface {
+	Anonymize(input string) (string, error)
+}
+
+type SqlAnonymizer struct {
+	registry IdentifierHasher
+}
+
+func NewSqlAnonymizer(registry IdentifierHasher) Anonymizer {
+	return &SqlAnonymizer{
+		registry: registry,
+	}
+}
+
+func (a *SqlAnonymizer) Anonymize(inputSql string) (string, error) {
+	parseResult, err := queryparser.Parse(inputSql) // Parse the input SQL to ensure it's valid
+	if err != nil {
+		return "", fmt.Errorf("error parsing input SQL: %w", err)
+	}
+
+	visited := make(map[protoreflect.Message]bool)
+	parseTreeMsg := queryparser.GetProtoMessageFromParseTree(parseResult)
+	err = queryparser.TraverseParseTree(parseTreeMsg, visited, a.anonymizationProcessor)
+	if err != nil {
+		return "", fmt.Errorf("error traversing parse tree: %w", err)
+	}
+
+	anonymizedSql, err := queryparser.DeparseParseTree(parseResult)
+	if err != nil {
+		return "", fmt.Errorf("error deparsing parse tree: %w", err)
+	}
+
+	return anonymizedSql, nil
+}
+
+func (a *SqlAnonymizer) anonymizationProcessor(msg protoreflect.Message) error {
+	// three categories of nodes: Identifier nodes, literal nodes, Misc nodes for rest
+	a.identifierNodesProcessor(msg)
```

Comment:
here we seem to be silently ignoring errors. let's check for errors here

Reply:
missed this. done